### PR TITLE
[chore]: Actions - Cache Dependencies

### PR DESCRIPTION
**Pull request by lukepistrol**
In order to cut down the time of our CI with Github Actions, dependencies are now cached using [actions/cache](https://github.com/actions/cache).

This action plugin checks `Package.resolved` and when there are no changes, it downloads the cached build folder instead of fetching each dependency one-by-one.

This saves about ~5-7 minutes per test run. Once there are changes in `Package.resolved`, of course, it will result in a longer test run.

Screen Shot 2022-10-04 at 20 30 23

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ let package = Package(
     dependencies: [
         .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
         .package(url: "https://github.com/krzyzanowskim/STTextView", exact: "0.0.20"),
-        .package(url: "https://github.com/ChimeHQ/SwiftTreeSitter", exact: "0.6.1"),
```
Comment (thecoolwinter) [inline Package.swift:17]: I might be wrong here but I think this was just changed to use the exact version?

**Review by thecoolwinter — CHANGES_REQUESTED** [review]
Left a comment :)

File: Package.swift (hunk 1 above)
Comment (lukepistrol) [reply Package.swift:17]: Oh thanks, this was for testing purposes only!

**Review by austincondiff — DISMISSED** [review]
This is awesome, good thinking here. That will save a lot of time!

**Comment by lukepistrol** [comment]
There seems to be an issue still. It worked without a problem on my fork. Have to investigate. Converting this to a Draft now.

**Comment by lukepistrol** [comment]
Ok it seems to work now.

Just for anyone curious: removing `.swiftpm` from `.gitignore` solved it. Reference: https://github.com/apple/swift-package-manager/issues/5767#issuecomment-1256427240

**Review by austincondiff — APPROVED** [review]

**Pull request merged**